### PR TITLE
Add `onDismiss` handler and `displayCloseButton` to `PaywallView` and `PaywallFooterView`

### PR DESCRIPTION
**Pull request by tonidero**
This PR adds support for the `onDismiss` callback to `PaywallView` and `PaywallFooterView`. Additionally, it also adds the `displayCloseButton` parameter to `PaywallView` to test it more easily.

### CODE REVIEW

File: purchases_ui_flutter/ios/Classes/PurchasesUiPaywallView.swift
Comment (tonidero) [inline purchases_ui_flutter/ios/Classes/PurchasesUiPaywallView.swift:62]: I'm holding this since in iOS, we will need https://github.com/RevenueCat/purchases-ios/pull/3738

**Review by vegaro — APPROVED** [review]

File: purchases_ui_flutter/lib/views/paywall_footer_view.dart
Comment (tonidero) [inline purchases_ui_flutter/lib/views/paywall_footer_view.dart:35]: Just noticed this was missing so I added before merging it. cc @vegaro

**Pull request merged**
